cognite/client/_api/simulators/models.py

```diff
@@ -0,0 +1,437 @@
+
+            Retrieve simulator model by external id:
+                >>> res = client.simulators.models.retrieve(external_ids=["model_external_id", "model_external_id2"])
+        """
+        self._warning.warn()
+        identifiers = IdentifierSequence.load(ids=ids, external_ids=external_ids)
+        return self._retrieve_multiple(
+            list_cls=SimulatorModelList,
+            resource_cls=SimulatorModel,
+            identifiers=identifiers,
+        )
+
+    def __iter__(self) -> Iterator[SimulatorModel]:
+        """Iterate over simulator models
+
+        Fetches simulator models as they are iterated over, so you keep a limited number of simulator models in memory.
+
+        Returns:
+            Iterator[SimulatorModel]: yields Simulator model one by one.
+        """
+        return self()
+
+    @overload
+    def __call__(
+        self, chunk_size: int, filter: SimulatorModelsFilter | None = None, limit: int | None = None
+    ) -> Iterator[SimulatorModel]: ...
+
+    @overload
+    def __call__(
+        self, chunk_size: None = None, filter: SimulatorModelsFilter | None = None, limit: int | None = None
+    ) -> Iterator[SimulatorModel]: ...
+
+    def __call__(
+        self, chunk_size: int | None = None, filter: SimulatorModelsFilter | None = None, limit: int | None = None
+    ) -> Iterator[SimulatorModel] | Iterator[SimulatorModelList]:
+        """Iterate over simulator simulator models
+
+        Fetches simulator models as they are iterated over, so you keep a limited number of simulator models in memory.
+
+        Args:
+            chunk_size (int | None): Number of simulator models to return in each chunk. Defaults to yielding one simulator model a time.
+            filter (SimulatorModelsFilter | None): Filter to apply on the model revisions list.
+            limit (int | None): Maximum number of simulator models to return. Defaults to return all items.
+
+        Returns:
+            Iterator[SimulatorModel] | Iterator[SimulatorModelList]: yields Simulator one by one if chunk is not specified, else SimulatorList objects.
+        """
+        return self._list_generator(
+            list_cls=SimulatorModelList,
+            resource_cls=SimulatorModel,
+            method="POST",
+            filter=filter.dump() if isinstance(filter, CogniteFilter) else filter,
+            chunk_size=chunk_size,
+            limit=limit,
+        )
+
+    @overload
+    def create(self, models: Sequence[SimulatorModel]) -> SimulatorModelList: ...
+
+    @overload
+    def create(self, models: SimulatorModel | SimulatorModelWrite) -> SimulatorModelList: ...
+
+    def create(
+        self, models: SimulatorModel | SimulatorModelWrite | Sequence[SimulatorModel] | Sequence[SimulatorModelWrite]
+    ) -> SimulatorModel | SimulatorModelList:
+        """`Create simulator models <https://developer.cognite.com/api#tag/Simulator-Models/operation/create_simulator_model_simulators_models_post>`_
```

Comment:
I won't comment on this more, but let's fix all occurrences of:
1. Only use Writer classes in create or apply methods
2. Overloads that handle single inputs should not have lists as outputs